esphome/components/esp32/preferences.cpp

```diff
@@ -97,12 +100,12 @@ class ESP32Preferences : public ESPPreferences {
   ESPPreferenceObject make_preference(size_t length, uint32_t type) override {
     auto *pref = new ESP32PreferenceBackend();  // NOLINT(cppcoreguidelines-owning-memory)
     pref->nvs_handle = nvs_handle;
-    current_offset += length;
+    // current_offset += length;                     // not needed
 
-    uint32_t keyval = current_offset ^ type;
+    uint32_t keyval = type;  // removed current_offset from calculation
```

Comment:
same here - the comment isn't useful anymore once the code is committed.
Applies to the all other change related comments as well. 
```suggestion
    uint32_t keyval = type;
```